### PR TITLE
Dropdown list item renderer

### PR DESCRIPTION
**Pull request by anthonykaras-sage**
Allowing dropdown list to use a function passed as a prop to render list items if prop present.
Otherwise option.name will be used

- [ ] release Notes

### CODE REVIEW

**Comment by SeanArmstrong** [comment]
Looks good to me. 

Will need some release notes adding but just need to work out what version it should target

File: src/components/dropdown/dropdown.js [hunk 1]
```diff
@@ -630,7 +630,7 @@ class Dropdown extends React.Component {
           onMouseOver={ this.handleMouseOverListItem }
           className={ klass }
         >
-          { option.name }
+          { this.props.renderItem ? this.props.renderItem(option) : option.name }
```
Comment (ianoxley) [inline src/components/dropdown/dropdown.js]: `renderItem` needs adding to the component's `propTypes` as an optional function.

Comment (anthonykaras-sage) [inline src/components/dropdown/dropdown.js]: Done and pushed @ianoxley.

File: src/components/dropdown/dropdown.js [hunk 2]
```diff
@@ -122,6 +122,14 @@ class Dropdown extends React.Component {
     readOnly: PropTypes.bool,
 
     /**
+     * An optional function to be passed that will render each of the dropdown's items.
+     *
+     * @property renderItem
+     * @type {Function}
+     */
+    renderItem: PropTypes.function,
```
Comment (andrewjtait) [inline src/components/dropdown/dropdown.js]: is `.function` valid? it's normally `.func`.

Comment (anthonykaras-sage) [inline src/components/dropdown/dropdown.js]: Well spotted @andrewjtait . I'll fix

**Pull request closed without merging**